modules/auth/proc-adding-gitlab-oauth-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,63 @@
+For local development::: `pass:[http://localhost:7007/api/auth/gitlab/handler/frame]`
+For a non-local deployment::: `pass:[http://{APP_FQDN}/api/auth/gitlab/handler/frame]`
++
+NOTE: Due to a peculiarity with GitLab OAuth, ensure the URL has no trailing `/` after 'frame'.
+*Scopes*:: Select the following scopes from the list, and validate:
++
+// NOTE: including here verbatim copy from Gitlab screen.
+`read_user`::: Grants read-only access to the authenticated user's profile through the /user API endpoint, which includes username, public email, and full name.
+Also grants access to read-only API endpoints under /users.
+`read_repository`::: Grants read-only access to the authenticated user's profile through the /user API endpoint, which includes username, public email, and full name.
+Also grants access to read-only API endpoints under /users.
+`write_repository`::: Grants read/write access to repositories on private projects using Git-over-HTTP (not using the API).
+`openid`::: Grants permission to authenticate with GitLab using OpenID Connect.
+Also gives read-only access to the user's profile and group memberships.
+`profile`::: Grants read-only access to the user's profile data using OpenID Connect.
+`email`::: Grants read-only access to the user's primary email address using OpenID Connect.
++
+Save the *Application ID* and *Secret* for the next step.
+
+. Set up the Gitlab authentication provider in the {product} configuration (`app-config.yaml`).
++
+Add the `gitlab` provider configuration to your `app-config.yaml` under the root `auth` configuration, and edit the configuration keys:
++
+.`app-config.yaml` fragment
+[source,yaml]
+----
+auth:
+  environment: development
+  providers:
+    gitlab:
+      development:
+        clientId: ${AUTH_GITLAB_CLIENT_ID} <1>
+        clientSecret: ${AUTH_GITLAB_CLIENT_SECRET} <2>
+        ## Uncomment and define when using self-hosted GitLab:
+        # audience: https://gitlab.company.com <3>
+        ## Uncomment and define when using a custom redirect URI
+        # callbackUrl: https://${BASE_URL}/api/auth/gitlab/handler/frame <3>
+## Uncomment to use Gitlab for sign-in:
+# signInPage: gitlab <5>
+----
+<1> `clientId`:: The *Application ID* that you generated on GitLab, such as
+`4928c033ab3d592845c044a653bc20583baf84f2e67b954c6fdb32a532ab76c9`.
+<2> `clientSecret`:: The *Secret* that you generated on Gitlab, such as `gloas-f2c9c350759cc08346fbf94a476ae83c579c76dd629fc5eeef9dc21eedfe0475`.
+<3> `audience` (optional):: When using a self-hosted Gitlab, enter your GitLab instance base URL, such as
+`pass:[https://gitlab.company.com]`.
+<4> `callbackUrl` (optional):: Enter the URL matching the *Redirect URI* registered when creating your GitLab OAuth App, such as:
+* For local development: `pass:[http://localhost:7007/api/auth/gitlab/handler/frame]`
+* For a non-local deployment: `pass:[http://{APP_FQDN}:{APP_BACKEND_PORT}/api/auth/gitlab/handler/frame]`
```

Comment:
@kadel @riginoommen Is it really optional? What is the default behavior?

Reply:
I don't think think its optional call back URL is a Must

Reply:
default befhaviour is provider redirect to the redirect URI

Reply:
Ack